azure-pipelines/build-and-publish-dotnet-tools.yml

```diff
@@ -0,0 +1,176 @@
+trigger: none
```

Comment:
Is there any reason you have this as a separate pipeline instead of just always generating the tool version for each build? Seems simpler just to do that.

Reply:
Currently we build the tool if the BUILD_DOTNET_TOOL environment variable is set, since we need different build properties for the tool vs the standalone application. The tool doesn't have a corresponding exe file in the resulting nupkg, for example, and we also need to exclude the Migration service for now since it's not supported on arm64 yet. I could try looking using build targets instead, or task-scoped variables.

Reply:
Didn't know I could set an environment variable on a single build task, so that fixes the whole issue. I'll get rid of the new build yaml.